src/libs/IOUUtils.js

```diff
@@ -63,7 +64,76 @@ function updateIOUOwnerAndTotal(iouReport, actorEmail, amount, currency, type =
 }
 
+/**
+ * Returns the list of IOU actions depnding on the type and whehter or not they are pending.
+ * Used below so that we can decide if an IOU report is pending currency conversion.
+ *
+ * @param {Array} reportIOUActions
+ * @param {Object} iouReport
+ * @param {String} type - iouReportAction type. Can be oneOf(create, decline, cancel, pay, split)
+ * @param {String} pendingAction
+ * @param {Boolean} filterRequestsInDifferentCurrency
+ *
+ * @returns {Array}
+ */
+function getIOUReportActions(reportIOUActions, iouReport, type, pendingAction = '', filterRequestsInDifferentCurrency = false) {
+    return _.chain(reportIOUActions)
+        .filter(action => action.originalMessage
+            && action.originalMessage.IOUReportID.toString() === iouReport.reportID.toString()
+            && action.originalMessage.type === type)
+        .filter(action => (!_.isEmpty(pendingAction) ? action.pendingAction === pendingAction : true))
+        .filter(action => (filterRequestsInDifferentCurrency ? action.originalMessage.currency !== iouReport.currency : true))
+        .value();
+}
+
+/**
+ * Returns whether or not an IOU report contains money requests in a different currency
+ * that are either created or cancelled offline, and thus haven't been converted to the report's currency yet
+ *
+ * @param {Array} reportActions
+ * @param {Object} iouReport
+ *
+ * @returns {Boolean}
+ */
+function isIOUReportPendingCurrencyConversion(reportActions, iouReport) {
+    // Pending money requests that are in a different currency
+    const pendingRequestsInDifferentCurrency = _.chain(getIOUReportActions(
+        reportActions,
+        iouReport,
+        CONST.IOU.REPORT_ACTION_TYPE.CREATE,
+        CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        true,
+    )).map(action => action.originalMessage.IOUTransactionID).value();
+
+    // Pending cancelled money requests that are in a different currency
+    const pendingCancelledRequestsInDifferentCurrency = _.chain(getIOUReportActions(
+        reportActions,
+        iouReport,
+        CONST.IOU.REPORT_ACTION_TYPE.CANCEL,
+        CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        true,
+    )).map(action => action.originalMessage.IOUTransactionID).value();
+
+    const hasPendingRequests = pendingRequestsInDifferentCurrency.length || pendingCancelledRequestsInDifferentCurrency.length;
+
+    // If we have pending requests, check if all of them have been cancelled offline, if yes, the IOU report is not pending currency conversion
+    if (hasPendingRequests && pendingCancelledRequestsInDifferentCurrency.length === pendingRequestsInDifferentCurrency.length) {
+        const areAllRequestsInDifferentCurrencyCancelled = _.every(
+            pendingCancelledRequestsInDifferentCurrency,
+            requestTransactionID => _.contains(pendingRequestsInDifferentCurrency, requestTransactionID),
+        );
+        if (areAllRequestsInDifferentCurrencyCancelled) {
+            return false;
+        }
```

Comment:
I think we could simplify this by calling underscore intersection on this https://underscorejs.org/#intersection such that if pendingCancelledRequestsInDifferentCurrency **_intersection_** pendingRequestsInDifferentCurrency length === pendingCancelledRequestsInDifferentCurrency.length that means it is proper subset hence the same.

Personally that is a bit easier but i like Venn diagrams so that might be why :D

Reply:
I simplified this even more by simply checking if both arrays are identical :D.

Reply:
e voila